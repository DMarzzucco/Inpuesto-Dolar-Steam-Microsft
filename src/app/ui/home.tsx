"use client";

import React from "react";
import { useLocalContext } from "../context/LocalContext";
import { AdverHome, HomeComponent } from "./ui.comps";

export const HomeComp: React.FC = () => {
    const { booleanState, ButtonBoolean } = useLocalContext()
    return (
        <div className="my-4 flex flex-col justify-center items-center">
            <HomeComponent />
            {!booleanState.card ?
                <button onClick={() => { ButtonBoolean({ type: "openCard" }) }}>Leer Antes</button>
                : <AdverHome />
            }
        </div>
    )
}