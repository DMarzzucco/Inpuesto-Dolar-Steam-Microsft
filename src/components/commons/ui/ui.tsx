import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSteam, faMicrosoft } from "@/assets"
import { MenuProps } from "@/ts";
import Link from "next/link";
import React from "react"


export default class UiComps {

    Menunav: React.FC = () => {
        return (
            <div className='flex flex-col justify-start items-start'>
                <this.LinkNav logo={faSteam} title='Steam' path={'/Steam'} />
                <this.LinkNav logo={faMicrosoft} title='Microsoft' path={'/Microsoft'} />
            </div>
        )
    }


    LinkNav: React.FC<MenuProps> = ({ path, title, logo }) => {
        return (
            <Link href={path} className=' w-200 flex flex-row justify-start items-center p-2 border rounded-xl my-1 hover:bg-slate-300 hover:text-slate-900'>
                <FontAwesomeIcon icon={logo} />
                <p className='mx-2'>{title}</p>
            </Link>
        )
    }
    
}