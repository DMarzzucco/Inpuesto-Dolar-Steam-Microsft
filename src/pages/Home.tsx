import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { AdverHome, HomeComp } from "../components/Comps.tsx";
import { faMicrosoft, faSteam } from "../components/icons.ts";
import { useState } from "react";

function Home() {
    const [abrirCard, setAbrirCard] = useState<boolean>(false);
    const abrirMenu = () => {
        setAbrirCard(!abrirCard);
    }
    const closeMenu = () => { setAbrirCard(false) }
    return (
        <>
            <section className="flex flex-col  justify-center items-center w-full h-screen">
                <div className="my-4 flex flex-col justify-center items-center">
                    <h1 className="my-3 text-xl">Calculadora <span className="text-blue-400">AR$</span> en</h1>
                    <div className="flex flex-row text-4xl">
                        <div className="flex flex-row items-center">
                            <FontAwesomeIcon className="mx-2" icon={faMicrosoft} />
                            <h2 className="border-r pr-3">Microsoft</h2>
                        </div>
                        <div className="flex flex-row items-center">
                            <FontAwesomeIcon className="pl-1 mx-2" icon={faSteam} />
                            <h2>Steam</h2>
                        </div>

                    </div>
                </div>
                <h1 className="text-2xl ">Elige tu plataforma</h1>
                <HomeComp />
                <button className={`${abrirCard ? 'hidden' : ''}`} onClick={abrirMenu}>Leer Antes</button>
                {abrirCard && <AdverHome onClick={closeMenu} />}
            </section>
        </>
    )

}

export default Home;