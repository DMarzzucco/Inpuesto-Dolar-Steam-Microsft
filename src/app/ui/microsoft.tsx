"use client";

import React from "react";
import { faMicrosoft } from "@/assets"
import { useLocalContext } from "../context/LocalContext";
import { ecuationProp } from "@/ts";
import { OpTemplate } from "./ui.comps";

export const Microsoft: React.FC = React.memo(() => {
    const { ...pre } = useLocalContext()
    const State: ecuationProp = {
        inputValue: pre.values.microsoft,
        factor: 1724.07,
        precision: 4
    }
    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        pre.ecuation(State, { type: "microsoft" })
    }
    return (<OpTemplate
        click={() => { pre.ButtonBoolean({ type: "closeMicrosoft" }) }}
        icon={faMicrosoft}
        color="sky-blue"
        Name="Microsoft"
        onSubmit={handleSubmit}
        value={pre.values.microsoft}
        nameInput="microsoft"
        result={pre.resultado}
        factor={State}
    />)
})
Microsoft.displayName = 'Microsoft'