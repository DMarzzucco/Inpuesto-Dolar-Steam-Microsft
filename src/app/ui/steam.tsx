"use client";

import React from "react";
import { faSteam } from "@/assets"
import { useLocalContext } from "../context/LocalContext";
import { ecuationProp } from "@/ts";
import { OpTemplate } from "./ui.comps";

export const Steam: React.FC = React.memo(() => {

    const { ...pre } = useLocalContext()

    const State: ecuationProp = {
        inputValue: pre.values.steam,
        factor: 1585.79,
        precision: 2
    }
    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        pre.ecuation(State, { type: "steam" })
    }

    return (<OpTemplate
        click={() => { pre.ButtonBoolean({ type: "closeSteam" }) }}
        icon={faSteam}
        color="gray"
        Name="Steam"
        onSubmit={handleSubmit}
        value={pre.values.steam}
        nameInput="steam"
        result={pre.resultado}
        factor={State}

    />);
})
Steam.displayName = 'Steam';