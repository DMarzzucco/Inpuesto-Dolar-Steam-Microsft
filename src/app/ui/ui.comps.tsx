"use client";
import { ecuationProp, ListProps, TemplateProp } from "@/ts"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

import { faSteam, faMicrosoft, faCircleXmark } from "@/assets"
import React from "react";
import { Steam } from "./steam";
import { Microsoft } from "./microsoft";
import { useLocalContext } from "../context/LocalContext";


export const HomeComponent: React.FC = () => {
    const { ...pre } = useLocalContext()
    return (
        <div className="flex flex-col justify-center items-center border border-slate-400 rounded-2xl my-2 px-2 py-4 " >
            <h1 className="text-2xl font-semibold ">Elige tu Plataforma</h1>
            <ul className="flex flex-col justify-center items-center p-3">
                <ListHm color='gray' logo={faSteam} click={() => { pre.ButtonBoolean({ type: "openSteam" }) }} title="Steam" />
                <ListHm color='skyblue' logo={faMicrosoft} click={() => { pre.ButtonBoolean({ type: "openMicrosoft" }) }} title="Microsoft" />
            </ul>
            {pre.booleanState.steam && <Steam />}
            {pre.booleanState.microsoft && <Microsoft />}
        </div>
    )
}
HomeComponent.displayName = "HomeComponent";

export const ListHm: React.FC<ListProps> = ({ ...pre }) => {
    return (
        <button onClick={pre.click}
            className="flex flex-row justify-start items-center hover:bg-white hover:text-slate-800 p-2 my-1 rounded-2xl border border-slate-600 md:w-300 w-200">
            <FontAwesomeIcon icon={pre.logo} color={pre.color} />
            <p className="mx-2">{pre.title}</p>
        </button>
    )
}
ListHm.displayName = "ListHm";

export const AdverHome: React.FC = () => {
    const { ButtonBoolean } = useLocalContext()
    return (
        <div className='absolute w-auto border border-red-400 bg-red-900 p-3 rounded-xl flex flex-col items-end '>
            <button className='' onClick={() => { ButtonBoolean({ type: "closeCard" }) }}>
                <FontAwesomeIcon icon={faCircleXmark} />
            </button>
            <div className='flex flex-col justify-center items-center'>
                <h1 className='font-bold p-1 text-xl my-3'>IMPORTANTE</h1>
                <p className='text-center'>
                    Los resultados obtenidos, son un aproximado cercano que se puede obtener. <br /> El total real
                    dependerá en sí de los impuestos implementados por el banco que se use.
                </p>
                <span className='font-bold my-3 text-center'>
                    Última Actualización <br /> {new Date().toLocaleDateString()}
                </span>
            </div>
        </div>
    )
}
AdverHome.displayName = "AdverHome";

export const ResultComp: React.FC<{ State: ecuationProp }> = ({ State }) => {
    const { resultado } = useLocalContext()
    return (
        <div className="border border-slate-400 bg-slate-700 rounded-2xl my-2 p-2">
            <div className='flex flex-col my-2 p-1'>
                <h3 className='text-slate-300 font-semibold'>Total de impuestos nacionales y provinciales:</h3>
                <p className='text-20 font-bold'>60,00%</p>
            </div>
            <div className='flex flex-col border-b my-2 p-1'>
                <h1 className="text-xl m-2 font-semibold">Coste del dolar Tarjeta:</h1>
                <div>
                    <div className="font-bold">
                        <span>1.00 U$D =</span>
                        <span>{State.factor} AR$</span>
                    </div>
                    <p className="font-semibold text-slate-300">
                        Valor de dolar oficial mas impuestos.
                    </p>
                    <p className="text-13 text-slate-400 font-semibold">
                        Update: {new Date().toLocaleDateString()}
                    </p>
                </div>
            </div>
            <div>
                <h2 className='font-bold'>TOTAL APROXIMADO: </h2>
                <h1 className='font-bold text-20'>{resultado} AR$</h1>
            </div>
        </div>
    )
}
ResultComp.displayName = "ResultComp";

export const ErrorComp: React.FC = () => {
    return (
        <div className="border border-red-300 bg-red-900 rounded-2xl my-2 p-2">
            <h1 className="text-center text-slate-300">Ingrese un valor correcto</h1>
        </div>
    )
}
ErrorComp.displayName = "ErrorComp";

export const OpTemplate: React.FC<TemplateProp> = React.memo(({ ...pre }) => {
    const { takeInput, booleanState } = useLocalContext()
    return (
        <div className=" flex flex-col justify-center items-center border border-slate-400 p-10 rounded-xl">
            <button onClick={pre.click} className="w-full flex -mt-2">
                <FontAwesomeIcon icon={faCircleXmark} />
            </button>
            <div className="flex flex-col justify-center items-center">
                <FontAwesomeIcon className="text-30" icon={pre.icon} color={pre.color} />
                <h1 className="text-xl m-2">{pre.Name}</h1>
            </div>
            <form onSubmit={pre.onSubmit} className="flex flex-col justify-center items-center">
                <input
                    type="text"
                    placeholder="$0.00"
                    value={pre.value}
                    onChange={takeInput}
                    name={pre.nameInput}
                />
                <button className="hover:bg-slate-200 hover:text-slate-800 py-1 px-2 border m-2 border-slate-400 rounded-xl" type="submit">
                    Enviar
                </button>
            </form>
            {pre.result !== null && <ResultComp State={pre.factor} />}
            {booleanState.error && <ErrorComp />}
        </div>
    )
})
OpTemplate.displayName = "OpTemplate";
