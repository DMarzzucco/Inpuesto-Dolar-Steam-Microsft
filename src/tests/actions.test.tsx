import { render, screen } from "@testing-library/react";
import AuthLocalContext from "@/app/context/LocalContext";
import { HomeComp } from "@/app/ui/home";
import Home from "@/app/page";

// Mock de los componentes
jest.mock("@/app/context/LocalContext", () => ({
  __esModule: true,
  default: jest.fn(({ children }) => <div>{children}</div>),
}));

jest.mock("@/app/ui/home", () => ({
  __esModule: true,
  HomeComp: jest.fn(() => <div>Home Component</div>),
}));

describe("Home component", () => {
  it("should render the Home component with AuthLocalContext and HomeComp", () => {
    render(<Home />);

    expect(screen.getByText("Home Component")).toBeInTheDocument();
  });
});
