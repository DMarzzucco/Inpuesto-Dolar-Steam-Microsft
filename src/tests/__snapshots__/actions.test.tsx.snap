// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page render homepage unchange 1`] = `
<div>
  <section
    class="flex flex-col  justify-center items-center w-full h-screen"
  >
    <div
      class="my-4 flex flex-col justify-center items-center"
    >
      <div
        class="flex flex-col justify-center items-center border border-slate-400 rounded-2xl my-2 px-2 py-4 "
      >
        <h1
          class="text-2xl font-semibold "
        >
          Elige tu Plataforma
        </h1>
        <ul
          class="flex flex-col justify-center items-center p-3"
        >
          <button
            class="flex flex-row justify-start items-center hover:bg-white hover:text-slate-800 p-2 my-1 rounded-2xl border border-slate-600 md:w-300 w-200"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-steam "
              color="gray"
              data-icon="steam"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 256c0 137-111.2 248-248.4 248-113.8 0-209.6-76.3-239-180.4l95.2 39.3c6.4 32.1 34.9 56.4 68.9 56.4 39.2 0 71.9-32.4 70.2-73.5l84.5-60.2c52.1 1.3 95.8-40.9 95.8-93.5 0-51.6-42-93.5-93.7-93.5s-93.7 42-93.7 93.5v1.2L176.6 279c-15.5-.9-30.7 3.4-43.5 12.1L0 236.1C10.2 108.4 117.1 8 247.6 8 384.8 8 496 119 496 256zM155.7 384.3l-30.5-12.6a52.79 52.79 0 0 0 27.2 25.8c26.9 11.2 57.8-1.6 69-28.4 5.4-13 5.5-27.3.1-40.3-5.4-13-15.5-23.2-28.5-28.6-12.9-5.4-26.7-5.2-38.9-.6l31.5 13c19.8 8.2 29.2 30.9 20.9 50.7-8.3 19.9-31 29.2-50.8 21zm173.8-129.9c-34.4 0-62.4-28-62.4-62.3s28-62.3 62.4-62.3 62.4 28 62.4 62.3-27.9 62.3-62.4 62.3zm.1-15.6c25.9 0 46.9-21 46.9-46.8 0-25.9-21-46.8-46.9-46.8s-46.9 21-46.9 46.8c.1 25.8 21.1 46.8 46.9 46.8z"
                fill="currentColor"
              />
            </svg>
            <p
              class="mx-2"
            >
              Steam
            </p>
          </button>
          <button
            class="flex flex-row justify-start items-center hover:bg-white hover:text-slate-800 p-2 my-1 rounded-2xl border border-slate-600 md:w-300 w-200"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-microsoft "
              color="skyblue"
              data-icon="microsoft"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 32h214.6v214.6H0V32zm233.4 0H448v214.6H233.4V32zM0 265.4h214.6V480H0V265.4zm233.4 0H448V480H233.4V265.4z"
                fill="currentColor"
              />
            </svg>
            <p
              class="mx-2"
            >
              Microsft
            </p>
          </button>
        </ul>
      </div>
      <button>
        Leer Antes
      </button>
    </div>
  </section>
</div>
`;
